### PR TITLE
Ffm 908

### PR DESCRIPTION
**Pull request by milos85vasic-harness**
This pull request contains the following changes:

- Metrics implementation for the Android (with basic unit tests)
- Feature configurations obtain mechanism
- Unit tests updates for easier future tests writing and extension
- Basic SDK documentation on how to use Metrics.

### CODE REVIEW

**Review by davejohnston — APPROVED** [review]
This looks good to me

File: README.md [hunk 1]
```diff
@@ -129,6 +129,29 @@ Following table provides summary on possible event types and corresponding respo
 To avoid unexpected behaviour, when listener is not needed anymore, a caller should call 
 `CfClient.getInstance().unregisterEventsListener(eventsListener)`. This way the sdk will remove desired listener from internal list.
 
+## _Using feature flags metrics_
+
+In order to turn on metrics for feature flags evaluations it is required to define your configuration like in the following example:
+```kotlin
+val remoteConfiguration = CfConfiguration.builder()
+            .enableStream(true)
```
Comment (rushabh-harness) [inline README.md:137]: this should be default configuration, no need to define it explicitly

File: README.md [hunk 2]
```diff
@@ -129,6 +129,29 @@ Following table provides summary on possible event types and corresponding respo
...
+            .pollingInterval(10)
+            .enableAnalytics(true)
```
Comment (rushabh-harness) [inline README.md:139]: same thing for analytics

File: README.md [hunk 3]
```diff
@@ -129,6 +129,29 @@ Following table provides summary on possible event types and corresponding respo
 To avoid unexpected behaviour, when listener is not needed anymore, a caller should call 
 `CfClient.getInstance().unregisterEventsListener(eventsListener)`. This way the sdk will remove desired listener from internal list.
 
+## _Using feature flags metrics_
+
+In order to turn on metrics for feature flags evaluations it is required to define your configuration like in the following example:
+```kotlin
+val remoteConfiguration = CfConfiguration.builder()
+            .enableStream(true)
+            .pollingInterval(10)
```
Comment (rushabh-harness) [inline README.md:138]: what is the default polling interval acc to playbook ?

File: cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java [hunk 4]
```diff
@@ -1,14 +1,32 @@
...
+import java.util.Map;
+import java.util.Set;
+
+import io.harness.cfsdk.utils.CfUtils;
+
 public class Target {
+
     private String name;
     private String identifier;
+    private boolean isPrivate; // If the target is private
```
Comment (rushabh-harness) [inline cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java:14]: There is only 1 target in the system so not sure what this means if the target is private

File: cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java [hunk 5]
```diff
@@ -1,14 +1,32 @@
...
+    private Set<String> privateAttributes; // Custom set to set the attributes which are private
```
Comment (rushabh-harness) [inline cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java:15]: Does it make sense to have private attributes in client SDK ?

**Review by rushabh-harness — CHANGES_REQUESTED** [review]
Please go over the comments I have listed in the PR. Let me know what you feel

**Comment by milos85vasic-harness** [comment]
@rushabh-harness I have addressed all of your comments since each of them Is valid. Please look at my latest changes (my last commit) and let me know once you finish the review. Thank you!

**Review by rushabh-harness — APPROVED** [review]

**Pull request merged**